docs/Announcement-Post.md

```diff
@@ -0,0 +1,269 @@
+```
+
+And you can call it like so:
+```python
+    q: NamedArray # [Batch, Pos, HeadDim]
+    k, v: NamedArray # [Batch, KPos, HeadDim]
+    attn = dot_product_attention(HeadDim, KPos, q, k, v)  # [Batch, Pos, HeadDim]
+```
+
+I find this code much more readable than the equivalent code using positional axes. Moreover, named axes are more
+flexible. Do you want multi-head dot product attention? You don't have to do anything, just call the function.
+Or maybe you want to, say, support dot product attention over image patches? you... still don't have to do anything, just
+call the function with two "Position" axes:
+
+```python
+    q: NamedArray # [Batch, Head, Height, Width, HeadDim]
+    k, v: NamedArray # [Batch, Head, KHeight, KWidth, HeadDim]
+    attn = dot_product_attention(HeadDim, (KHeight, KWidth), q, k, v)  # [Batch, Head, Height, Width, HeadDim]
+```
+
+Moreover, named axes have yet another advantage: they make it even easier to scale up to large models. In particular,
+we can use named axes (and Jax's `pjit` sharding) to implement tensor parallelism and FSDP in a way that is easy to use.
+With Levanter and Haliax, FSDP is as simple as specifying an axis (typically the "embed" or "hidden" axis for transformers)
+that you want to keep your parameters and optimizer states sharded along. We then distinguish between two "configurations"
+of your parameters: one for storage (`parameter_axis_mapping`) and one for compute (`compute_axis_mapping`).
+
+For example, here's what your training script might look like:
+
+
+```python
+  def compute_loss(model, data):
+    ...
+
+
+  @named_pjit(axis_resources=trainer.parameter_axis_mapping)
+  def train_step(opt_state, model, data):
+      with hax.axis_mapping(trainer.compute_axis_mapping):
+          loss, grads = jax.value_and_grad(compute_loss)(model, data)
+      updates, opt_state = optimizer.update(grads, opt_state, params=model)
+      model = eqx.apply_updates(model, updates)
+      return loss, model, opt_state
+```
+
+Under the hood, all we do is use Jax's `pjit` operator to shard the parameters and optimizer states along the axis
+you specify. We then use `hax.axis_mapping` to change the preferred mapping for computing the loss and gradients. Jax
+handles most of the heavy lifting, except Linear layers need a bit of handholding, which Haliax provides.
+
+Tensor Parallelism works in a broadly similar manner, with you specifying how "wide" you want model parallelism to be
+and specifying which axes of your model should be sharded. (For a transformer, this would typically be the Head
+and MLP axes.)
+
+In our experiments, we have found that, for models of the sizes that we can reasonably train with our resources, FSDP
+is more than enough to get good scaling, especially on TPU. In practice, with FSDP, we haven't found tensor parallelism
+to be useful for training transformers to at least 20B parameters. (Tensor parallelism is useful for inference, but
```

Comment:
This sentence is bit hard to understand, but I don't have better idea to rewrite it.

So I asked chatgpt, here's its suggestion:
> Through our experiments, we discovered that Fully Sharded Data Parallel (FSDP) provides sufficient scaling for the model sizes we can feasibly train using our available resources, particularly on Tensor Processing Units (TPU). In practical applications, using FSDP, we observed that tensor parallelism offered limited utility in training transformers with up to 20 billion parameters.

Reply:
thanks!